compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -88,11 +88,28 @@ loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
 
       for (uint32_t axis = 0; axis < shape_by_input.rank(); ++axis)
       {
-        shape_by_input.dim(axis) = const_shape_node->at<S32>(axis);
         if (const_shape_node->at<S32>(axis) < 0)
         {
           shape_by_input.dim(axis).unset();
         }
+        else if (const_shape_node->at<S32>(axis) == 0)
+        {
+          const auto node_tensor = loco::must_cast<luci::CircleNode *>(node->tensor());
+          // set dim value to input
+          if (node_tensor->shape_status() == luci::ShapeStatus::VALID && axis < node_tensor->rank())
+            shape_by_input.dim(axis) = node_tensor->dim(axis);
+          else
+          {
+            // stop to check if this case exist for debugging
+            LUCI_ASSERT(false, "Check Reshape shape with 0");
+          }
+        }
+        else
+        {
+          shape_by_input.dim(axis).set(const_shape_node->at<S32>(axis));
+        }
+        // check valid or stop for debugging
+        assert(shape_by_input.dim(axis).value() > 0 || !shape_by_input.dim(axis).known());
```

Comment:
How about using `LUCI_ASSERT` as a unified assert?

```suggestion
        LUCI_ASSERT(shape_by_input.dim(axis).value() > 0 || !shape_by_input.dim(axis).known(), "Invalid input dim");
```